lib/sqlsync-solid-js/src/hooks.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  ConnectionStatus,
+  DocId,
+  DocType,
+  ParameterizedQuery,
+  QuerySubscription,
+  Row,
+  SQLSync,
+  normalizeQuery,
+  pendingPromise,
+} from "@orbitinghail/sqlsync-worker";
+import { Accessor, createEffect, createSignal, onCleanup, useContext } from "solid-js";
+import { SQLSyncContext } from "./context";
+
+export function useSQLSync(): Accessor<SQLSync> {
+  const [sqlSync] = useContext(SQLSyncContext);
+  if (import.meta.env.DEV && !sqlSync()) {
+    throw new Error(
+      "could not find sqlsync context value; please ensure the component is wrapped in a <SqlSyncProvider>"
+    );
+  }
+
+  return () => {
+    const sqlSyncValue = sqlSync();
+    if (import.meta.env.DEV && !sqlSyncValue) {
+      throw new Error(
+        "could not find sqlsync context value; please ensure the component is wrapped in a <SqlSyncProvider>"
+      );
+    } else if (!sqlSyncValue) {
+      console.error(
+        "could not find sqlsync context value; please ensure the component is wrapped in a <SqlSyncProvider>"
+      );
+    }
+    // biome-ignore lint/style/noNonNullAssertion: asserts in dev
+    return sqlSyncValue!;
+  };
+}
+
+type MutateFn<M> = (mutation: M) => Promise<void>;
+type UseMutateFn<M> = (docId: DocId) => MutateFn<M>;
+
+type UseQueryFn = <R = Row>(
+  docId: Accessor<DocId>,
+  query: Accessor<ParameterizedQuery | string>
+) => Accessor<QueryState<R>>;
+
+type SetConnectionEnabledFn = (enabled: boolean) => Promise<void>;
+type UseSetConnectionEnabledFn = (docId: DocId) => SetConnectionEnabledFn;
+
+export interface DocHooks<M> {
+  useMutate: UseMutateFn<M>;
+  useQuery: UseQueryFn;
+  useSetConnectionEnabled: UseSetConnectionEnabledFn;
+}
+
+export function createDocHooks<M>(docType: Accessor<DocType<M>>): DocHooks<M> {
+  const useMutate = (docId: DocId): MutateFn<M> => {
+    const sqlsync = useSQLSync();
+    return (mutation: M) => sqlsync().mutate(docId, docType(), mutation);
+  };
+
+  const useQueryWrapper = <R = Row>(
+    docId: Accessor<DocId>,
+    query: Accessor<ParameterizedQuery | string>
+  ) => {
+    return useQuery<M, R>(docType, docId, query);
+  };
+
+  const useSetConnectionEnabledWrapper = (docId: DocId) => {
+    const sqlsync = useSQLSync();
+    return (enabled: boolean) => sqlsync().setConnectionEnabled(docId, docType(), enabled);
+  };
+
+  return {
+    useMutate,
+    useQuery: useQueryWrapper,
+    useSetConnectionEnabled: useSetConnectionEnabledWrapper,
+  };
+}
+
+export type QueryState<R> =
+  | { state: "pending"; rows?: R[] }
+  | { state: "success"; rows: R[] }
+  | { state: "error"; error: Error; rows?: R[] };
+
+export function useQuery<M, R = Row>(
+  docType: Accessor<DocType<M>>,
+  docId: Accessor<DocId>,
+  rawQuery: Accessor<ParameterizedQuery | string>
+): Accessor<QueryState<R>> {
+  const sqlsync = useSQLSync();
+  const [state, setState] = createSignal<QueryState<R>>({ state: "pending" });
+
+  createEffect(() => {
+    let query = normalizeQuery(rawQuery());
```

Comment:
This can be a const. Also can you run biome on this PR? I'll add it to just soon, but in the meantime you can run:

```
pnpm exec biome check ./lib/sqlsync-solid-js/src
```

Reply:
"let" haha, i've written too much rust. Ran biome. Thanks for the tip. Would you mind adding instructions on biome to the `CONTRIBUTING.md`?